tests/impl.cpp

```diff
@@ -257,6 +257,28 @@ result_t do_mm_store_ps(int32_t *p, int32_t x, int32_t y, int32_t z, int32_t w)
     return TEST_SUCCESS;
 }
 
+result_t test_mm_storeu_si16(const SSE2NEONTestImpl &impl, uint32_t i)
+{
+    const int32_t *_a = (const int32_t *) impl.mTestIntPointer1;
+    __m128i b;
+    __m128i a = do_mm_load_ps(_a);
+    _mm_storeu_si16(&b, a);
+    int16_t *_b = (int16_t *) &b;
+    int16_t *_c = (int16_t *) &a;
+    return validateInt16(a, _b[0], _c[1], _c[2], _c[3], _c[4], _c[5], _c[6],
```

Comment:
Although the result is correct, the logic is kind of weird.
It should be `validateInt16(b, _a[0], b[1], ......)`.
We care about the data stored into the memory instead of the input data for testing.